### DISCUSSION TITLE
How do I analyze the output with a test?

### DISCUSSION THREAD

**Discussion by tylers-username** [c4spar/cliffy#468] [Q&A]
I have a simple command that outputs a string to the Console. How do I capture that output and test that it is accurate?

For example, the action below, with the `--pipe` option, will output https://dummy.sh/. `.parse()` doesn't seem to return anything useful for me to asset that the returned URL equals the expected URL. 🤔 

```
    .action(({pipe, browser}, ...search) => {
        let path = '';
        if(search.length>0){
            path = `search.html?q=${search.join('%20')}`
        }
        const docsPath = `https://dummy.sh/${path}`;

        console.log(docsPath);

        if(!pipe) {
            openBrowserTo(docsPath)
        }
    });
```

```
# Test
import { assertEquals } from "https://deno.land/std@0.160.0/testing/asserts.ts";
import DocsCommand from "./DocsCommand.ts";

Deno.test("--pipe outputs URL to stdout",  async() => {
    console.log(await DocsCommand.parse(['--pipe']))
});

```

**Comment by c4spar** [comment]
Currently there is no build-in solution for testing.
For now you could spawn a child-process which executes your command and than read the output from the child process.
